app/controllers/geocoder_controller.rb

```diff
@@ -206,42 +206,51 @@ def normalize_params
-
-      elsif latlon = query.match(%r{^(?<lat>[+-]?\d+(?:\.\d+)?)(?:\s+|\s*[,/]\s*)(?<lon>[+-]?\d+(?:\.\d+)?)$})
-        params.merge!(:lat => latlon["lat"], :lon => latlon["lon"]).delete(:query)
-
+      if match = query.match(/^(?<ns>[NS])\s*#{dms_regexp('ns')}\W*(?<ew>[EW])\s*#{dms_regexp('ew')}$/) ||
+                 query.match(/^#{dms_regexp('ns')}\s*(?<ns>[NS])\W*#{dms_regexp('ew')}\s*(?<ew>[EW])$/)
+        captures = match.named_captures.compact
+        params.merge! :lat => dms_to_decdeg("ns", "ns", captures),
+                      :lon => dms_to_decdeg("ew", "ew", captures)
+        params.delete(:query)
+
+      elsif match = query.match(%r{^
+                      (?<ns>[+-]?)\s*#{dms_regexp('ns', :comma => false)}
+                      (?:\s+|\s*[,/]\s*)
+                      (?<ew>[+-]?)\s*#{dms_regexp('ew', :comma => false)}
+                    $}x)
+        captures = match.named_captures.compact
+        params.merge! :lat => dms_to_decdeg("ns", "+-", captures),
+                      :lon => dms_to_decdeg("ew", "+-", captures)
+        params.delete(:query)
         params[:latlon_digits] = true
       end
     end
 
     params.permit(:query, :lat, :lon, :latlon_digits, :zoom, :minlat, :minlon, :maxlat, :maxlon)
   end
 
-  def dms_regexp(name_prefix)
+  def dms_regexp(prefix, comma: true)
+    final_fraction = comma ? /(?:[.,]\d+)?/ : /(?:[.]\d+)?/
     /
-      (?: (?<#{name_prefix}d>\d{1,3}(?:\.\d+)?)°? ) |
-      (?: (?<#{name_prefix}d>\d{1,3})°?\s*(?<#{name_prefix}m>\d{1,2}(?:\.\d+)?)['′]? ) |
-      (?: (?<#{name_prefix}d>\d{1,3})°?\s*(?<#{name_prefix}m>\d{1,2})['′]?\s*(?<#{name_prefix}s>\d{1,2}(?:\.\d+)?)["″]? )
+      (?: (?<#{prefix}d>\d{1,3}#{final_fraction})°? ) |
+      (?: (?<#{prefix}d>\d{1,3})°?\s*(?<#{prefix}m>\d{1,2}#{final_fraction})['′]? ) |
+      (?: (?<#{prefix}d>\d{1,3})°?\s*(?<#{prefix}m>\d{1,2})['′]?\s*(?<#{prefix}s>\d{1,2}#{final_fraction})["″]? )
     /x
   end
 
-  def to_decdeg(captures)
-    ns = captures.fetch("ns").casecmp?("s") ? -1 : 1
-    nsd = BigDecimal(captures.fetch("nsd", "0"))
-    nsm = BigDecimal(captures.fetch("nsm", "0"))
-    nss = BigDecimal(captures.fetch("nss", "0"))
-
-    ew = captures.fetch("ew").casecmp?("w") ? -1 : 1
-    ewd = BigDecimal(captures.fetch("ewd", "0"))
-    ewm = BigDecimal(captures.fetch("ewm", "0"))
-    ews = BigDecimal(captures.fetch("ews", "0"))
-
-    lat = ns * (nsd + (nsm / 60) + (nss / 3600))
-    lon = ew * (ewd + (ewm / 60) + (ews / 3600))
-
-    { :lat => lat.round(6).to_s("F"), :lon => lon.round(6).to_s("F") }
+  def dms_to_decdeg(prefix, directions, captures)
+    extract_number = ->(suffix) { captures.fetch("#{prefix}#{suffix}", "0").sub(",", ".") }
+
+    positive_direction, negative_direction = directions.chars
+    sign = captures.fetch(prefix, positive_direction).casecmp?(negative_direction) ? "-" : ""
+    deg = if captures["#{prefix}m"] || captures["#{prefix}s"]
```

Comment:
Why do we need to special case the case when there are only degrees? What was wrong with the old way of doing it?

Reply:
Because we don't need to do min/sec math with divisions and results that have to be rounded to some precision. This special case was previously here: https://github.com/openstreetmap/openstreetmap-website/blob/1fce0c00f12ac6bbb56ad7c05c8c9f53c0d6edb3/app/controllers/geocoder_controller.rb#L213-L214 except it was triggered not by a lack of min/sec but by the entire query being in a simplest format possible. So you had [1.1111111111 2.2222222222](https://www.openstreetmap.org/search?query=1.1111111111%202.2222222222) keeping lat/lon intact but [1.1111111111 N 2.2222222222 E](https://www.openstreetmap.org/search?query=1.1111111111%20N%202.2222222222%20E) going into math mode with rounding.